component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,106 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.dpop.handler;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpHeaders;
+import org.wso2.carbon.identity.auth.service.AuthenticationContext;
+import org.wso2.carbon.identity.auth.service.AuthenticationRequest;
+import org.wso2.carbon.identity.auth.service.AuthenticationResult;
+import org.wso2.carbon.identity.auth.service.AuthenticationStatus;
+import org.wso2.carbon.identity.auth.service.exception.AuthenticationFailException;
+import org.wso2.carbon.identity.auth.service.handler.AuthenticationHandler;
+import org.wso2.carbon.identity.auth.service.util.AuthConfigurationUtil;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.dpop.constant.DPoPConstants;
+import org.wso2.carbon.identity.dpop.validators.DPoPValidator;
+import org.wso2.carbon.identity.oauth2.OAuth2TokenValidationService;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2ClientApplicationDTO;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2TokenValidationRequestDTO;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2TokenValidationResponseDTO;
+
+/**
+ * DPoPAuthenticationHandler will validate the requests authorized with DPoP access tokens.
+ */
+public class DPoPAuthenticationHandler extends AuthenticationHandler {
+
+    private static final Log log = LogFactory.getLog(DPoPAuthenticationHandler.class);
+
+    @Override
+    protected AuthenticationResult doAuthenticate(MessageContext messageContext) throws
+            AuthenticationFailException {
+
+        AuthenticationResult authenticationResult = new AuthenticationResult(AuthenticationStatus.FAILED);
+        AuthenticationContext authenticationContext = (AuthenticationContext) messageContext;
+        AuthenticationRequest authenticationRequest = authenticationContext.getAuthenticationRequest();
+        if (authenticationRequest != null) {
+
+            String authorizationHeader = authenticationRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            if (StringUtils.isNotBlank(authorizationHeader) &&
+                    authorizationHeader.startsWith(DPoPConstants.OAUTH_DPOP_HEADER) ) {
+                String accessToken;
+                String[] bearerToken = authorizationHeader.split(" ");
+                if (bearerToken.length == 2) {
+                    accessToken = bearerToken[1];
+                    OAuth2TokenValidationService oAuth2TokenValidationService = new OAuth2TokenValidationService();
+                    OAuth2TokenValidationRequestDTO requestDTO = new OAuth2TokenValidationRequestDTO();
+                    OAuth2TokenValidationRequestDTO.OAuth2AccessToken token = requestDTO.new OAuth2AccessToken();
+                    token.setIdentifier(accessToken);
+                    token.setTokenType(DPoPConstants.OAUTH_HEADER);
```

Comment:
why the header is set as Bearer here?

Reply:
if the tokenType is not set it will throw a NullPointerException from [ findAccessTokenValidator](https://github.com/wso2-support/identity-inbound-auth-oauth/blob/support-6.4.2/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java#L587) method when calling the [findOAuthConsumerIfTokenIsValid](https://github.com/kalanika/identity-oauth-addons/blob/feature-dpop-rest/component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java#L81) method.